### DISCUSSION TITLE
Is the implementation of the ipo algorithm in the project based on a single constraint?

### DISCUSSION THREAD

**Discussion by moodybluesf** [PKU-Alignment/omnisafe#257] [Q&A]
Hello! Thank you very much for your contribution to this project for safe reinforcement learning!
As a newcomer to security reinforcement learning, I would like to ask a question. Is the implementation of the ipo algorithm in the project based on a single constraint? If so, how should it be extended to the multi-constraint situation?

**Comment by Gaiejj** [comment]
We notice that this question is the same as the one we have answered in #258, so we close it.
